### ISSUE TITLE
Frotz 2.54 SHA256 hash is wrong

### ISSUE THREAD

**Issue by DavidGriffith** [Homebrew/homebrew-core#127318]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
I'm the upstream for Frotz.

I've been alerted that the SHA256 hash for Frotz 2.54 is wrong.  See https://gitlab.com/DavidGriffith/frotz/-/issues/272 for the report.  Would someone please tell me where that hash came from and when?  I'm investigating what could have gone wrong.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Trying to figure out why you have a bad SHA256 hash for the Frotz tarball.

### What happened (include all command output)?

SHA256 mismatch.  See above.

### What did you expect to happen?

Frotz to install.  See above.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
`brew install frotz`
```

**Comment by DavidGriffith** [comment]
After going over things, I cannot explain why the hash here differs from what is observed at Gitlab.  I have confirmed that the hash there (756d7e11370c9c8e61573e350e2a5071e77fd2781be74c107bd432f817f3abc7) matches what I have on my local machines for the 2.54 tag.

@Porkepix, would you please tell me where you got the hash for Frotz 2.54 in #99273?

**Comment by Porkepix** [comment]
I didn't get the hash, `brew bump` or `brew bump-formula-pr` got it, and fetched it directly from GitLab.
If the hash changed, you can open a PR to fix it to the new one.

**Comment by carlocab** [comment]
That was the hash for the tarball when it was downloaded and built from source in https://github.com/Homebrew/homebrew-core/pull/99273.

**Comment by cho-m** [comment]
May be a GitLab issue given a similar issue in #127055.

However, `svt-av1` is fine now and may have been related to https://gitlab.com/gitlab-org/gitlab/-/issues/402616 which was fixed a few days ago.

**Comment by qianbinbin** [comment]
Anyone has the copies of tarballs?

[Tarballs of `svt-av1`](https://github.com/Homebrew/homebrew-core/files/11114134/old-and-new.zip) have different directory names, `SVT-AV1-018276d714ce65d9b586f6205ee016cbd8d5425d` and `SVT-AV1-v1.4.1`, that's why they got different checksums, but the contents are exactly the same.

**Comment by cho-m** [comment]
Could also be a side-effect of `git` update in GitLab servers. Hard to pin down exact cause without debugging further.

I think both GitHub and GitLab have potentially unstable checksums for archive tarballs due to their design and currently require attaching download asset to remain unchanged. Usually not a problem, but they crop up every once in a while.

Anyway, if new SHA remains, we can merge #127298.

**Comment by chenrui333** [comment]
I just merged #127298, I think we can close this issue now.

**Comment by cho-m** [comment]
Closing for now as nothing left to do on Homebrew side.

Let us know if there are still issues.

---

On side note, we could consider switching to tarball from website (https://ifarchive.org/if-archive/infocom/interpreters/frotz/frotz-2.54.tar.gz), which may avoid issues from GitLab updates.

In this cause, we would probably want to mirror https://ifarchive.org/indexes/if-archive/infocom/interpreters/frotz/old/ to avoid breaking downloads when a new release is out that we didn't update to yet.

Can consider this change on a future release.
